File: Foundation/Scratch/String/1basics.java
public class basics{
    public static void test1(){
        String str = "abcde";  //O(n)
        String str1 = str;    //O(1)
        str += 'g';    //O(n)
        char ch = str.charAt(3);
        System.out.println(str);
    } 
    public static void test3(){
        StringBuilder sb = new StringBuilder();
        sb.append('d');   //O(1)
        StringBuilder sb1 = sb;    //O(1)
        StringBuilder sb2 = new StringBuilder(sb);     //O(n)
        char ch = sb.charAt(0);
        System.out.println(sb.toString());   //O(n)
    }
    public static void main(String[] args){
        String str = scn.nextLine();
        test1();
        test3();
    }
}

File: Basics/Strings/1basics.cpp
#include<iostream>
using namespace std;
void test1(){
    string str = "abcde";  //O(n)
    str += 'f';    //O(1)
    str.push_back('g');    //O(1)
    cout<<str<<endl;
    string str1 = str;   //O(n)
    str1 += 'p';    //O(1)
    str = str +'p';     //O(n)
    char ch = str[4];
    cout<<str<<" "<<str1<<endl;    //O(n)
}
int main(){
    string str;
    cin>>str;
    test1();
    return 0;
}